protocol/app/ante/gas_test.go

```diff
@@ -132,3 +138,67 @@ func TestValidateMsgType_FreeInfiniteGasDecorator(t *testing.T) {
 		})
 	}
 }
+
+func TestSubmitTxnWithGas(t *testing.T) {
+	tests := map[string]struct {
+		gasFee   sdk.Coins
+		expectOk bool
+	}{
+		"Success - 5 cents usdc gas fee": {
+			gasFee:   constants.TestFeeCoins_5Cents,
+			expectOk: true,
+		},
+		"Success - 5 cents native token gas fee": {
+			gasFee:   constants.TestFeeCoins_5Cents_NativeToken,
+			expectOk: true,
+		},
+		"Failure: 0 gas fee": {
+			gasFee:   sdk.Coins{},
+			expectOk: false,
+		},
+		"Failure: unsupported gas fee denom": {
+			gasFee: sdk.Coins{
+				// 1BTC, which is not supported as a gas fee denom, and should be plenty to cover gas.
+				sdk.NewCoin(constants.BtcUsd.Denom, sdkmath.NewInt(100_000_000)),
+			},
+			expectOk: false,
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			msg := &bank.MsgSend{
+				FromAddress: constants.BobAccAddress.String(),
+				ToAddress:   constants.AliceAccAddress.String(),
+				Amount: []sdk.Coin{
+					sdk.NewCoin(assets.AssetUsdc.Denom, sdkmath.NewInt(1)),
+				},
+			}
+
+			tApp := testapp.NewTestAppBuilder().
+				WithAppCreatorFn(
+					testapp.DefaultTestAppCreatorFn(map[string]interface{}{},
+						baseapp.SetMinGasPrices(cmd.MinGasPrice),
+					)).
+				WithTesting(t).
+				Build()
+			ctx := tApp.InitChain()
+
+			msgSendCheckTx := testapp.MustMakeCheckTxWithPrivKeySupplier(
+				ctx,
+				tApp.App,
+				testapp.MustMakeCheckTxOptions{
+					AccAddressForSigning: constants.BobAccAddress.String(),
+					Gas:                  constants.TestGasLimit,
+					FeeAmt:               tc.gasFee,
+				},
+				constants.GetPrivateKeyFromAddress,
+				msg,
+			)
+
+			checkTx := tApp.CheckTx(msgSendCheckTx)
+			// Sanity check that gas was used.
+			require.Greater(t, checkTx.GasUsed, int64(0))
+			require.Equal(t, tc.expectOk, checkTx.IsOK())
```

Comment:
Nit: is it possible to do a more specific check (error message?) than just `Ok`?